wemake_python_styleguide/visitors/ast/function_empty_lines.py

```diff
@@ -8,53 +9,124 @@
 
 
 @final
-class WrongEmptyLinesCountVisitor(base.BaseTokenVisitor):
-    """Restricts empty lines in function or method body."""
+class _Function(object):
 
-    def __init__(self, *args, **kwargs) -> None:
-        """Initializes a counter."""
-        super().__init__(*args, **kwargs)
-        self._empty_lines_count = 0
-        self._function_start_line = 0
+    def __init__(self, file_tokens: Tuple[tokenize.TokenInfo, ...]):
+        self._tokens = file_tokens
 
-    def visit(self, token: tokenize.TokenInfo) -> None:
-        """Find empty lines count."""
-        self._try_mark_function_start(token)
-        if self._function_start_line:
-            if token.type == tokenize.NL and token.line == '\n':
-                self._empty_lines_count += 1
-            self._check_empty_lines(token)
-
-    def _check_empty_lines(self, token: tokenize.TokenInfo):
-        if token.type == tokenize.DEDENT and token.line == '':
-            func_lines = token.start[0] - self._function_start_line - 1
-            if self._empty_lines_count:
-                available_empty_lines = self._available_empty_lines(
-                    func_lines,
-                    self._empty_lines_count,
+    def name_token(self) -> tokenize.TokenInfo:
+        return self._tokens[1]
+
+    def body(self) -> str:
+        target_tokens = []
+        for token in self._tokens:
+            if self._is_target_line(token):
+                continue
+            target_tokens.append(
+                token,
+            )
+        return ''.join([target_token.string for target_token in target_tokens])
+
+    def _is_target_line(self, token) -> bool:
+        is_comment = '#' in token.line
+        is_string = token.type == tokenize.STRING
+        is_multistring_end = '"""' in token.line
+        return is_comment or is_string or is_multistring_end
+
+
+@final
+class _FileFunctions(object):
+
+    def __init__(self, file_tokens: Tuple[tokenize.TokenInfo, ...]):
+        self._file_tokens = file_tokens
+
+    def as_list(self) -> List[_Function]:
+        functions = []
+        function_tokens: List[tokenize.TokenInfo] = []
+        in_function = False
+        for token in self._file_tokens:
+            if self._is_definition_token(token):
+                in_function = True
+            elif self._is_function_end(token, function_tokens):
+                in_function = False
+                functions.append(_Function(tuple(function_tokens)))
+                function_tokens = []
+            if in_function:
+                function_tokens.append(token)
+        return functions
+
+    def _is_definition_token(self, token) -> bool:
```

Comment:
I think that `_is_definition_token` and `_is_function_end` should be named: `_is_function_start` and `_is_function_end`

So we can clearly tell that they are the same thing.

Reply:
Yes, I'm agree. Changed